src/odb/src/db/dbITerm.cpp

```diff
@@ -813,6 +813,14 @@ std::vector<dbAccessPoint*> dbITerm::getPrefAccessPoints() const
   return aps;
 }
 
+void dbITerm::clearPrefAccessPoints()
+{
+  _dbITerm* iterm = (_dbITerm*) this;
+  // Clear aps_ map instead of destroying dbAccessPoint object to prevent
+  // destroying APs of other iterms.
+  iterm->aps_.clear();
```

Comment:
Are the APs shared somehow?

Reply:
From my tests, it seems so. When I destroyed the APs of the instances that were moved or had their masters swapped, a bunch of other instances that weren't changed at all lost their APs.

If I understand it correctly, the APs are associated to the dbMPin, under dbMTerm. So all the dbITerms that share the same dbMTerm would lost their APs if we delete it from one of them.

Reply:
In that case we should eventually reference count the APs so they can be cleaned up on the last inst.  That and undo can wait for a future PR.